helm/cloud-provider-azure/templates/cloud-provider-azure.yaml

```diff
@@ -0,0 +1,399 @@
+            {{- end }}
+            - "--route-reconciliation-period={{ .Values.cloudControllerManager.routeReconciliationPeriod }}"
+            - "--port={{ .Values.cloudControllerManager.port }}"
+            {{- if hasKey .Values.cloudControllerManager "profiling" }}
+            - "--profiling={{ .Values.cloudControllerManager.profiling }}"
+            {{- end }}
+            {{- if hasKey .Values.cloudControllerManager "securePort" }}
+            - "--secure-port={{ .Values.cloudControllerManager.securePort }}"
+            {{- end }}
+            {{- if hasKey .Values.cloudControllerManager "useServiceAccountCredentials" }}
+            - "--use-service-account-credentials={{ .Values.cloudControllerManager.useServiceAccountCredentials }}"
+            {{- end }}
+            - "--v={{ .Values.cloudControllerManager.logVerbosity }}"
+          resources:
+            requests:
+              cpu: {{ .Values.cloudControllerManager.containerResourceManagement.requestsCPU }}
+              memory: {{ .Values.cloudControllerManager.containerResourceManagement.requestsMem }}
+            limits:
+              cpu: {{ .Values.cloudControllerManager.containerResourceManagement.limitsCPU }}
+              memory: {{ .Values.cloudControllerManager.containerResourceManagement.limitsMem }}
+          livenessProbe:
+            httpGet:
+              path: /healthz
+              port: {{ .Values.cloudControllerManager.port }}
+            initialDelaySeconds: 20
+            periodSeconds: 10
+            timeoutSeconds: 5
+          volumeMounts:
+            - name: etc-kubernetes
+              mountPath: /etc/kubernetes
+            - name: etc-ssl
+              mountPath: /etc/ssl
+              readOnly: true
+            - name: msi
+              mountPath: /var/lib/waagent/ManagedIdentity-Settings
+              readOnly: true
+      volumes:
+        - name: etc-kubernetes
+          hostPath:
+            path: /etc/kubernetes
+        - name: etc-ssl
+          hostPath:
+            path: /etc/ssl
+        - name: msi
+          hostPath:
+            path: /var/lib/waagent/ManagedIdentity-Settings
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  labels:
+    k8s-app: cloud-node-manager
+  name: cloud-node-manager
+  namespace: kube-system
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cloud-node-manager
+  labels:
+    k8s-app: cloud-node-manager
+rules:
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["watch", "list", "get", "update", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: cloud-node-manager
+  labels:
+    k8s-app: cloud-node-manager
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: cloud-node-manager
+subjects:
+  - kind: ServiceAccount
+    name: cloud-node-manager
+    namespace: kube-system
+---
+apiVersion: apps/v1
```

Comment:
could you also include the Daemonset for Windows nodes?

Reply:
Sure thing, is this the right reference to use?

https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/examples/out-of-tree/cloud-node-manager.yaml#L102-L155

Reply:
I think so.